doc/lms.mysql

```diff
@@ -2557,6 +2558,21 @@ CREATE TABLE customercontacts (
 		FOREIGN KEY (customerid) REFERENCES customers (id) ON DELETE CASCADE ON UPDATE CASCADE
 ) ENGINE=InnoDB;
 
+# --------------------------------------------------------
+#
+# Structure of table "customercontactproperties"
+#
+CREATE TABLE customercontactproperties (
+    contactid   int(11)         NOT NULL,
+    name        varchar(255)    NOT NULL DEFAULT '',
+    value       varchar(255)    NOT NULL DEFAULT '',
```

Comment:
A może dla name i value nie pozwalać na NULL, a jednocześnie nie dawać DEFAULT '' - będzie trudniej o pomyłkę w kodzie, bo SQL zareaguje natychmiast?

Reply:
Zgadza się. Chciałem tak od razu zrobić, ale coś mnie wtedy od tego odwiodło tylko teraz nie mogę sobie przypomnieć co :)
Już wiem:
`$propertyName = Utils::removeInsecureHtml($property['name']);
$propertyValue = Utils::removeInsecureHtml($property['value']);'
to mogło zwrócić wartość pusty ciąg. Tak było na początku jak pisałem poprawkę, ale teraz w ostatecznej wersji już jest to w kodzie zablokowane w funkcji `validate_customer_emails` i w przypadku pustego ciągu w ogóle nie dochodzi do próby wstawiania rekordu do 'customercontactproperties'.

Zaraz puszczę kolejną poprawkę zgodnie z Twoją sugestią.